cmd/p2p/sensor/sensor.go

```diff
@@ -386,44 +386,49 @@ func handleAPI(server *ethp2p.Server, counter *prometheus.CounterVec) {
 	}
 }
 
-// Function to handle DNS discovery and peer addition
-func runDNSDiscovery(server *ethp2p.Server, peers map[enode.ID]string, peersMutex *sync.Mutex) {
-    if inputSensorParams.DiscoveryDNS != "" {
-        log.Info().Msgf("Starting DNS discovery sync from %s", inputSensorParams.DiscoveryDNS)
-        
-        dnsclient := dnsdisc.NewClient(dnsdisc.Config{})
-        tree, err := dnsclient.SyncTree(inputSensorParams.DiscoveryDNS)
-        if err != nil {
-            log.Error().Err(err).Msg("Failed to sync DNS discovery tree")
-            return
-        }
-        
-        // Log the number of nodes in the tree
-        log.Info().Msgf("Successfully synced DNS discovery tree. Found %d unique nodes", len(tree.Nodes()))
-
-        // Lock the peers map and server operations
-        peersMutex.Lock()
-        defer peersMutex.Unlock()
-
-        // Add DNS-discovered peers
-        for _, node := range tree.Nodes() {
-            if _, ok := peers[node.ID()]; !ok {
-                peers[node.ID()] = node.URLv4()
-                
-                log.Debug().Msgf("Discovered new peer: %s", node.URLv4())
-
-                // Write peers to nodes file
-                if err := p2p.WritePeers(inputSensorParams.NodesFile, peers); err != nil {
-                    log.Error().Err(err).Msg("Failed to write DNS-discovered peers to file")
-                }
-
-                // Instruct server to connect to the new peer
-                server.AddTrustedPeer(node)
```

Comment:
I think this was the problem, I think `AddTrustedPeer` has to wait a little for the next "peering" cycle, where as `AddPeer` will add the peer immediately.